### PR TITLE
Fix missing camera ptr for Reconstruction.DeleteAllPoints2DAndPoints3D()

### PR DESCRIPTION
**Pull request by B1ueber2y**
Addresses https://github.com/colmap/colmap/issues/2775. Missed from https://github.com/colmap/colmap/pull/2762

### CODE REVIEW

**Comment by ahojnnes** [comment]
Is it possible to construct a test case that covers the issue? Thanks for the fix.

**Comment by B1ueber2y** [comment]
> Is it possible to construct a test case that covers the issue? Thanks for the fix.

Added some tests in the reconstruction class (we need to add some pipeline tests in the future). PTAL when time permits.  The point_triangulator breaks in the current head so we should merge this PR ASAP : )

File: src/colmap/scene/reconstruction.cc [hunk 1]
```diff
@@ -43,6 +43,15 @@ namespace colmap {
 
 Reconstruction::Reconstruction() : max_point3D_id_(0) {}
 
+bool Reconstruction::Check() const {
+  for (const auto& image : images_) {
```
Comment (ahojnnes) [inline src/colmap/scene/reconstruction.cc:47]: I recommend moving this check to the test file, where its only use case is. No need to clutter the public interface of the reconstruction.

**Review by ahojnnes — COMMENTED** [review]
Thanks, small suggestion. Otherwise, LGTM.

File: src/colmap/scene/reconstruction.cc (hunk 1 above)
Comment (B1ueber2y) [reply src/colmap/scene/reconstruction.cc]: Done.

**Review by ahojnnes — APPROVED** [review]

**Review by ahojnnes — APPROVED** [review]

**Pull request merged**